File: first-quarter/introduction-to-programming/truthiness-and-falsiness/truthiness-and-falsiness.py
# This file is an example of how truthiness and falsiness work in Python.

# Boolean values are the two constant objects False and True.
# Although the examples here cast to bool explicitly, it is rarely necessary to do so explicitly.
# Numbers are false if zero, and true if non-zero.
print(bool(0))  # False
print(bool(1))  # True

# None is always false.
print(bool(None))  # False

# Containers are false if empty, and true if non-empty.
print(bool([]))         # False
print(bool([1, 2, 3]))  # True

print(bool({}))         # False
print(bool({1, 2, 3}))  # True

File: first-quarter/introduction-to-programming/truthiness-and-falsiness/truthiness-and-falsiness.ts
// This file is an example of how truthiness and falsiness work in TypeScript.

// For numbers, 0 is falsy and all other numbers are truthy.
console.log(0 ? 'truthy' : 'falsy') // falsy
console.log(1 ? 'truthy' : 'falsy') // truthy

// For strings, empty strings are falsy and all other strings are truthy.
console.log('' ? 'truthy' : 'falsy')      // falsy
console.log('hello' ? 'truthy' : 'falsy') // truthy

// For arrays, empty arrays are falsy and all other arrays are truthy.
console.log([] ? 'truthy' : 'falsy')  // falsy
console.log([1] ? 'truthy' : 'falsy') // truthy

// For objects, all objects are truthy.
console.log({} ? 'truthy' : 'falsy') // truthy

// For keywords, undefined, null, and NaN are falsy.
console.log(undefined ? 'truthy' : 'falsy') // falsy
console.log(null ? 'truthy' : 'falsy')      // falsy
console.log(NaN ? 'truthy' : 'falsy')       // falsy
